File: Compare Version Numbers.py
class Solution:
	# @param version1, a string
	# @param version2, a string
	# @return an integer
	def compareVersion(self, version1, version2):
		nums1 = version1.split('.')
		nums2 = version2.split('.')
		for i in xrange(max(len(nums1), len(nums2))):
			ver1 = 0
			if (i < len(nums1)):
				ver1 = int(nums1[i])
			ver2 = 0
			if (i < len(nums2)):
				ver2 = int(nums2[i])
			if (ver1 < ver2):
				return -1
			if (ver1 > ver2):
				return 1
		return 0

File: Compare Version Numbers.java
public class Solution {
    public int compareVersion(String version1, String version2) {
        String[] nums1 = version1.split("\\.");
        String[] nums2 = version2.split("\\.");
        int i = 0;
        while (i < nums1.length || i < nums2.length) {
            int ver1 = 0;
            if (i < nums1.length)
                ver1 = Integer.valueOf(nums1[i]);
            int ver2 = 0;
            if (i < nums2.length)
                ver2 = Integer.valueOf(nums2[i]);
            if (ver1 < ver2)
                return -1;
            if (ver1 > ver2)
                return 1;
            i++;
        }
        return 0;
    }
}